flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerTest.java

```
@@ -1074,6 +1080,79 @@ protected void run() {
+				getTestActor(),
+				leaderSessionID);
+
+			try {
+				final IntermediateDataSetID resultId = new IntermediateDataSetID();
+
+				// Create the JM
+				ActorRef jm = system.actorOf(Props.create(
+					new SimplePartitionStateLookupJobManagerCreator(leaderSessionID, getTestActor())));
+
+				jobManager = new AkkaActorGateway(jm, leaderSessionID);
+
+				final int dataPort = NetUtils.getAvailablePort();
+				Configuration config = new Configuration();
+				config.setInteger(ConfigConstants.TASK_MANAGER_DATA_PORT_KEY, dataPort);
+				config.setInteger(TaskManagerOptions.NETWORK_REQUEST_BACKOFF_INITIAL, 100);
+				config.setInteger(TaskManagerOptions.NETWORK_REQUEST_BACKOFF_MAX, 200);
+				config.setString(ConfigConstants.TASK_MANAGER_LOG_PATH_KEY, "/i/dont/exist");
+
+				taskManager = TestingUtils.createTaskManager(
+					system,
+					jobManager,
+					config,
+					false,
+					true);
+
+				// ---------------------------------------------------------------------------------
+
+				final ActorGateway tm = taskManager;
+				final ExecutionContextExecutor context = ExecutionContext$.MODULE$.fromExecutor(Executors.newSingleThreadExecutor());
+
+				new Within(d) {
+					@Override
+					protected void run() {
+						Future<Object> logFuture = tm.ask(TaskManagerMessages.getRequestTaskManagerLog(), timeout);
+
+						logFuture.onSuccess(new OnSuccess<Object>() {
+								@Override
+								public void onSuccess(Object result) throws Throwable {
+									Assert.fail();
+								}
+							}, context);
+						logFuture.onFailure(new OnFailure() {
+							@Override
+							public void onFailure(Throwable failure) throws Throwable {
+								testActorGateway.tell(new Status.Success("success"));
+							}
+						}, context);
+						
+						Status.Success msg = expectMsgClass(Status.Success.class);
+						Assert.assertEquals("success", msg.status());
+					}
+				};
+			}
+			catch(Exception e) {
+				e.printStackTrace();
+				fail(e.getMessage());
```

Better let the exception bubble up. Less code ;-)